File: problem_16.pl
#!/usr/bin/env perl

sub add_digit{
	$total=0;
	$j = 0;
	$n = $_[0];
	while ($n > 0){
		$d[$j] = $n % 10;
		$n = ($n - $d[$j])/10;
		$j++;
		}
	foreach $p (reverse @d){
		print "$p";
		$total+=$p;
	}
	print "\n";
	print "TOTAL: $total\n";
}

add_digit(2**1000);

print 2**1000 ."\n";
print (2**1000)%100000000000 ."\n";

File: problem_16.py
#!/usr/bin/env python

def add_digit(n):
	d=[0] * 500
	total=0
	j=0
	p=0
	
	
	while n>0:
		d[j] = n % 10
		n = (n - d[j]) / 10
		j=j+1
	
	for p in d:
		total = total + p
	
	print "total = ", total
	



add_digit(2**1000)